src/workerd/io/hibernation-manager.h

```diff
@@ -151,6 +152,10 @@ class HibernationManagerImpl final : public Worker::Actor::HibernationManager {
     // Stores the last received autoResponseRequest timestamp.
     kj::Maybe<kj::Date> autoResponseTimestamp;
 
+    // Keeps track of the currently ongoing websocket auto-response send promise. This promise may
+    // be moved to api::websocket if an hibernating websocket unhibernates.
+    kj::Maybe<kj::Promise<void>> maybeAutoResponsePromise;
```

Comment:
Worth considering https://github.com/cloudflare/workerd/pull/1130#discussion_r1330613635 here as well, although I'm less certain because we are potentially moving this promise elsewhere so `Maybe` could be more appropriate?